default.nix

```diff
@@ -129,6 +129,78 @@ let
       ${nix2container-bin}/bin/nix2container image-from-dir $out ${dir}
     '';
 
+  pullImageByManifest =
+    { imageName
+    , imageManifest
+    # The manifest dictates what is pulled; these three are only used for
+    # the supplied manifest-updating scripts.
+    , imageTag ? "latest"
+    , os ? "linux"
+    , arch ? pkgs.go.GOARCH
+    , tlsVerify ? true
+    , registryUrl ? "registry.hub.docker.com"
```

Comment:
This URL works for both image-pulling and manifest-fetching, and I've validated that it works on my Harbor instance as well, returning an OCI image.

Reply:
It's a bit of a matter of taste, but I wonder if instead of having a separate parameter for `registryUrl`, if this should be optionally part of `imageName`, and we detect and act on it as necessary—basically just divide on slashes and assuming the first part is the URL if three parts are present. I expect that's effectively what `dockertools.pullImage` is doing, since it just passes through the `imageName` to skopeo, which encodes all these assumptions.

Reply:
I agree with you it would be better to get the registry URL from the image name.
But IIRC, the algorithm is a bit more complex since there is some magic with the `library` namespace.

Actually, i think we should use the `containers/image` library to write the getManifest script and the script which pulls layers. We could then rely on this library to parse the image name for us.

In the context of this PR, i would be ok to keep the `registryUrl` (and the bash implementation).

Reply:
Sounds good. It would be a straightforward matter later on for someone skilled in the art to translate these things over to Go.